src/app/modules/dashboard/components/dashboard/dashboard.component.ts

```diff
@@ -23,22 +25,25 @@ export class DashboardComponent implements OnInit {
     }
 
     ngOnInit() {
-        if (this.profileService.profile) {
-            this.profileService.getSavedSearches(this.profileService.profile.userId).subscribe(x => {
+        if (localStorage.getItem('profile')) {
+            this.profile = JSON.parse(localStorage.getItem('profile'));
+        };
+        if (this.profile) {
+            this.profileService.getSavedSearches(this.profile.userId).subscribe(x => {
                 this.logger.debug('saved searches received: {}', x.length);
                 this.savedSearches = x;
             });
-            this.profileService.getWatchedDatasets(this.profileService.profile.userId).subscribe(x => {
+            this.profileService.getWatchedDatasets(this.profile.userId).subscribe(x => {
                 this.logger.debug('saved searches received: {}', x.length);
                 this.watchedDatasets = x;
             });
         } else {
-            this.profileService.onProfileReceived.subscribe(x => {
+            this.profileService.getProfile().subscribe(x => {
                 this.profileService.getSavedSearches(x.userId).subscribe(r => {
```

Comment:
I don't think we need if else here. Because if user not logged, he'd not be here